### PR TITLE
Create private epinio CA

### LINKED ISSUE #603: Switch from self signed issuer to (private) CA issuer
This is a followup of #564

Instead of creating a self signed issuer (https://github.com/epinio/epinio/blob/main/deployments/cert_manager.go#L247) we should better create a CA issuer instead: https://cert-manager.io/docs/configuration/selfsigned/#bootstrapping-ca-issuers

See discussion here why: https://github.com/epinio/epinio/issues/476

### PR DESCRIPTION
**Pull request by andreas-kupries**
fix #603 

Merge should wait for #590 to be merged, and then rebased.

### CODE REVIEW

**Comment by andreas-kupries** [comment]
Made wip again ... The app cert signer is still self-signed, not the private CA I expected ... Investigating ...

**Comment by andreas-kupries** [comment]
Ok. Definitely broken, and getting the switch to it as default actually done exposed it.
The issuer itself is apparently not getting ready 

```
apiVersion: cert-manager.io/v1
kind: ClusterIssuer
metadata:
  creationTimestamp: "2021-07-14T09:50:25Z"
  generation: 1
  name: epinio-ca
  resourceVersion: "15037"
  uid: 4e916a47-3acc-4849-bb60-866b20f955a2
spec:
  ca:
    secretName: epinio-ca-root
status:
  conditions:
  - lastTransitionTime: "2021-07-14T09:51:16Z"
    message: 'Error getting keypair for CA issuer: certificate is not a CA'
    reason: ErrInvalidKeyPair
    status: "False"
    type: Ready
```

**Comment by andreas-kupries** [comment]
References

  - https://github.com/jetstack/cert-manager/issues/2497
  - https://cert-manager.io/docs/configuration/#supported-issuer-types
  - https://docs.cert-manager.io/en/release-0.8/tasks/issuers/setup-ca.html
 
... Ah :facepalm: and :scream: ... I seem to have forgotten the line `isCA: true` for the root cert when I pulled things from https://cert-manager.io/docs/configuration/selfsigned/#bootstrapping-ca-issuers

**Comment by andreas-kupries** [comment]
And with that fix it works.

File: deployments/cert_manager.go [hunk 1]
```diff
@@ -207,7 +208,76 @@ func (cm CertManager) apply(ctx context.Context, c *kubernetes.Cluster, ui *term
...
+	// With the self signed issuer in place it is now possible to bootstrap
+	// Epinio's private CA. Phase 1, the CA root certificate, signed by self
+	// signed.
+
+	caCert := fmt.Sprintf(`{
+		"apiVersion" : "cert-manager.io/v1alpha2",
+		"kind"       : "Certificate",
+		"metadata"   : {
+			"name"      : "epinio-ca",
+			"namespace" : "%s"
```
Comment (manno) [inline deployments/cert_manager.go:225]: nit: this is not needed when using a namespaced client, like in 256

**Review by manno — APPROVED** [review]

File: deployments/cert_manager.go (hunk 1 above)
Comment (andreas-kupries) [reply deployments/cert_manager.go:225]: Will fix.

Comment (andreas-kupries) [reply deployments/cert_manager.go]: Done, commit 0c9dd5aea7099f744bfc8571d32ec374e7e243c6

**Review by manno — APPROVED** [review]

**Pull request merged**